GDJS/Runtime/runtimeobject.ts

```diff
@@ -730,6 +778,168 @@ namespace gdjs {
       return this._variables.has(name);
     }
 
+    /**
+     * Returns the collection of effects to be rendered
+     * by the underlying renderer.
+     * @returns The render effects.
+     */
+    getRendererEffects() {
+      return this._rendererEffects;
+    }
+
+    initializeEffect(effectsData: EffectData) {
+      const layer = this._runtimeScene.getLayer(this.layer);
+      this._runtimeScene
+        .getGame()
+        .getObjectEffectsManager()
+        .initializeEffect(effectsData, this._rendererEffects, layer);
```

Comment:
We should not need to pass the layer?

Reply:
Oh, this is definitely a mistake. But I've one doubt here, we definitely don't need a layer here, but an `EffectsTarget` (which is either a runtime object or a layer). So, we'll still need to pass the runtime object (instead of a layer), but I guess that should still be very expensive right?

Reply:
Well it depends :) What do you mean exactly by "this should still be very expensive"?
If we pass the RuntimeObject, i.e: `this`, it's a reference that is passed like all objects in JavaScript, so internally it's just basically passing a pointer as an argument, which is very cheap :) 

It seems fair to pass to something called "initializeEffect": the effects data, the map where to store effects and the target (i.e: the runtime object) (which can be queried for its width/height for example).